gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -319,12 +317,12 @@ public String toString() {
     return "Fork: TaskId = \"" + this.taskId + "\" Index: \"" + this.index + "\" State: \"" + this.forkState + "\"";
   }
 
+  /**
+   * Tell this fork that the parent task is done. This is a second chance call if the parent
+   * task failed and didn't do so through the normal way of calling markParentTaskDone().
+   */ 
   @Override
   public void close() throws IOException {
-    // Tell this fork that the parent task is done. This is a second chance call if the parent
-    // task failed and didn't do so through the normal way of calling markParentTaskDone().
-    this.parentTaskDone = true;
```

Comment:
With my current patch, the fork does not keep the state of the ParentTask anymore it will just signal itself to shutdown. The markParentTaskDone method itself only fails when the Task gets interrupted. We could keep state if we did signal already and only signal once, wich is I think the best thing todo

I quickly need to check what this implies all the logic in Task,processRecord()
It looks like one stalled fork will stall the whole task as before, wich is wanted behaviour i guess, and my patch is not changing to much of it, except that fork 1 can't see a record n before fork0 anymore, but fork1 still can't see record n+1 before fork 0 saw record n. Maybe this whole logic can also be removed then. I think the gain to keep it is very small.